### ISSUE TITLE
Redudant replacement of private_service_access

### ISSUE THREAD

**Issue by szandala** [terraform-google-modules/terraform-google-sql-db#448]
### TL;DR

Whenever I do `terraform plan` using module `private_service_access` I receive notification that network has changed and requires recreation of resource. The network is NOT changing!

### Expected behavior

No changes to be applied

### Observed behavior

```
 # module.private-service-access-default.google_compute_global_address.google-managed-services-range must be replaced
-/+ resource "google_compute_global_address" "google-managed-services-range" {
      ~ address            = "..." -> (known after apply)
      ~ creation_timestamp = "2023-03-27T15:43:34.673-07:00" -> (known after apply)
      ~ id                 = "projects/mmstage/global/addresses/google-managed-services-default" -> (known after apply)
      ~ label_fingerprint  = "...=" -> (known after apply)
      - labels             = {} -> null
        name               = "google-managed-services-default"
      ~ network            = "https://www.googleapis.com/compute/v1/projects/mmstage/global/networks/default" -> (known after apply) # forces replacement
      ~ self_link          = "https://www.googleapis.com/compute/v1/projects/mmstage/global/addresses/google-managed-services-default" -> (known after apply)
        # (4 unchanged attributes hidden)
    }

  # module.private-service-access-default.google_service_networking_connection.private_service_access must be replaced
-/+ resource "google_service_networking_connection" "private_service_access" {
      ~ id                      = "https%3A%2F%2Fwww.googleapis.com%2Fcompute%2Fv1%2Fprojects%...%2Fglobal%2Fnetworks%2Fdefault:servicenetworking.googleapis.com" -> (known after apply)
      ~ network                 = "https://www.googleapis.com/compute/v1/projects/.../global/networks/default" -> (known after apply) # forces replacement
      ~ peering                 = "servicenetworking-googleapis-com" -> (known after apply)
        # (2 unchanged attributes hidden)
    }
```

### Terraform Configuration

```hcl
module "private-service-access-default" {
  source      = "GoogleCloudPlatform/sql-db/google//modules/private_service_access"
  project_id  = var.project_id
  vpc_network = "default"
  labels      = {}
  depends_on = [
    module.gke_private_cluster,
  ]
}
```
```


### Terraform Version

```sh
terraform version
Terraform v1.3.7
on linux_amd64
+ provider registry.terraform.io/cloudflare/cloudflare v4.2.0
+ provider registry.terraform.io/hashicorp/google v4.58.0
+ provider registry.terraform.io/hashicorp/google-beta v4.58.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.19.0
+ provider registry.terraform.io/hashicorp/null v3.2.1
+ provider registry.terraform.io/hashicorp/random v3.4.3
```
```


### Additional information

_No response_

**Comment by ravisiddhu** [comment]
I tried to re-create the issue, but for me 'terraform plan' is running successfully and not producing any terraform drift. Could you please provide some more context?

**Comment by icy95** [comment]
Hi @ravisiddhu exact same issue happens while trying to adding an api to `activate_apis` in `terraform-google-modules/project-factory/google` module in same terraform state. And there is no need to even apply the `activate_api` changes, just adding another api to the configuration and running `terraform plan` is enough to produce the same exact changes in the first comment.

**Comment by jstuckey** [comment]
@szandala did you ever resolve your issue? We are seeing the same diff when trying to remove some temporary subnets that we created for testing.
